File: recipes/java-recipes/spring-boot-postgres-recipe/src/main/java/com/trendyol/stove/examples/java/spring/application/external/category/CategoryHttpApi.java
package com.trendyol.stove.examples.java.spring.application.external.category;

import com.trendyol.stove.recipes.shared.application.BusinessException;
import com.trendyol.stove.recipes.shared.application.category.CategoryApiResponse;

import reactor.core.publisher.Mono;

public interface CategoryHttpApi {
    Mono<CategoryApiResponse> getCategoryById(int id) throws BusinessException;
}

File: recipes/kotlin-recipes/ktor-mongo-recipe/src/main/kotlin/com/trendyol/stove/examples/kotlin/ktor/application/external/CategoryHttpApi.kt
package com.trendyol.stove.examples.kotlin.ktor.application.external

import com.trendyol.stove.recipes.shared.application.category.CategoryApiResponse

interface CategoryHttpApi {
  suspend fun getCategory(id: Int): CategoryApiResponse
}

data class CategoryApiConfiguration(
  val url: String,
  val timeout: Long
)
